src/network/networkd-nexthop.c

```diff
@@ -458,9 +458,32 @@ static int nexthop_acquire_id(Manager *manager, NextHop *nexthop) {
         return -EBUSY;
 }
 
-void log_nexthop_debug(const NextHop *nexthop, const char *str, Manager *manager) {
-        _cleanup_free_ char *state = NULL, *group = NULL, *flags = NULL;
+static int nexthop_to_string(const NextHop *nexthop, Manager *manager, char **ret) {
+        _cleanup_free_ char *group = NULL, *flags = NULL;
+
+        assert(nexthop);
+        assert(manager);
+        assert(ret);
+
+        (void) route_flags_to_string_alloc(nexthop->flags, &flags);
+
         struct nexthop_grp *nhg;
+        HASHMAP_FOREACH(nhg, nexthop->group)
+                (void) strextendf_with_separator(&group, ",", "%"PRIu32":%"PRIu32, nhg->id, nhg->weight+1u);
+
+        char *str;
+        if (asprintf(&str, "id: %"PRIu32", gw: %s, blackhole: %s, group: %s, flags: %s",
```

Comment:
btw you can just pass `ret` as first param, no need for the `str` indirection...

the asprintf() docs never mention this explicitly, but it's a GNU extension and the implementation in glibc never initializes the parameter on failure, hence matches our usual semantics...